include/ipc/ipc_service.h

```diff
@@ -170,6 +170,80 @@ int ipc_service_register_endpoint(const struct device *instance,
  */
 int ipc_service_send(struct ipc_ept *ept, const void *data, size_t len);
 
+/** @brief Get an empty TX buffer to be sent using @ref ipc_service_send_nocopy
+ *
+ *  This function can be called to get an empty TX buffer so that the
+ *  application can directly put its data into the sending buffer without copy
+ *  from an application buffer.
+ *
+ *  It is the application responsibility to correctly fill the allocated TX
+ *  buffer with data and passing correct parameters to @ref
+ *  ipc_service_send_nocopy function to perform data no-copy-send mechanism.
+ *
+ *  @param ept Registered endpoint by @ref ipc_service_register_endpoint.
+ *  @param data Pointer to the empty TX buffer.
+ *  @param len Pointer to store the TX buffer size.
+ *  @param wait Boolean, wait or not for buffer to become available
+ *
+ *  @retval -EIO when no backend is registered or send hook is missing from
+ *	    backend.
+ *  @retval -EINVAL when instance or endpoint is invalid.
+ *  @retval -ENOTSUP when this is not supported by backend.
+ *  @retval -ENOBUFS when there are no TX buffers available.
+ *
+ *  @retval 0 on success.
+ *  @retval other errno codes depending on the implementation of the backend.
+ */
+int ipc_service_get_buffer(struct ipc_ept *ept, void **data, uint32_t *len, int wait);
```

Comment:
nit: docs state that `wait` type is boolean
```suggestion
int ipc_service_get_buffer(struct ipc_ept *ept, void **data, uint32_t *len, bool wait);
```

Reply:
would it make sense to have `k_timeout_t` here?

Reply:
ok